src/framework/operations.hpp

```diff
@@ -491,6 +491,9 @@ inline Op make_multiplexer(const reg_t &qubits,
   if (1ULL << num_controls != num_mats) {
     throw std::invalid_argument("invalid number of multiplexer matrices.");
   }
+  if (num_controls == 0 && mats.size() == 1) {
+    return make_unitary(qubits, mats[0]);
+  }
```

Comment:
I'm not sure I understand what's special about there only being a single matrix here?  It feels like either the existing loop should be updated to work with `num_controls=0`, or this return value should have some special logic to multiplex several matrices without controls.  (Updating the regular loop seems like the more likely choice.)

But it may just be that I don't fully understand what the code is doing.

Reply:
if `num_controls` is `0`, a size of matrices must be `1` because `num_controls` is defined with `std::log2(num_mats)`.

I simplified the guard condition to enter this special case.